src/block/compress.rs

```diff
@@ -379,10 +383,11 @@ pub(crate) fn compress_internal<T: HashTable, const USE_DICT: bool>(
     assert!(ext_dict.len() <= super::WINDOW_SIZE);
     assert!(ext_dict.len() <= input_stream_offset);
     if USE_DICT {
+        // Check for overflow hazard when using ext_dict
         assert!(input_stream_offset
             .checked_add(input.len())
             .and_then(|i| i.checked_add(ext_dict.len()))
-            .map_or(false, |i| i <= usize::MAX / 2));
+            .map_or(false, |i| i <= usize::MAX));
```

Comment:
`i <= usize::MAX`
this evaluates always to true

Reply:
Good point, I shouldn't have changed this.